nginx-controller/nginx/configurator.go

```diff
@@ -68,6 +68,66 @@ func (cnf *Configurator) addOrUpdateIngress(ingEx *IngressEx) {
 	cnf.ingresses[name] = ingEx
 }
 
+// AddOrUpdateIngress adds or updates NGINX configuration for the Ingress resources with Mergeable Types
+func (cnf *Configurator) AddOrUpdateMergableIngress(mergeableIngs MergableIngresses) error {
+	master := mergeableIngs.Master.Ingress
+
+	cnf.addOrUpdateMergableIngress(mergeableIngs, master.Name)
+
+	if err := cnf.nginx.Reload(); err != nil {
+		return fmt.Errorf("Error when adding or updating ingress %v/%v: %v", master.Namespace, master.Name, err)
+	}
+	return nil
+}
+
+func (cnf *Configurator) addOrUpdateMergableIngress(mergeableIngs MergableIngresses, name string) {
+	nginxCfg := IngressNginxConfig{}
+	masterServer := Server{}
+	locations := []Location{}
+	upstreams := []Upstream{}
+	keepalive := ""
+
+	master := mergeableIngs.Master
+	pems, jwtKeyFileName := cnf.updateSecrets(&master)
+	nginxCfg = cnf.generateNginxCfg(&master, pems, jwtKeyFileName)
+	masterServer = nginxCfg.Servers[0]
+	masterServer.IngressResource = name
+
+	for _, val := range nginxCfg.Upstreams {
+		upstreams = append(upstreams, val)
+	}
+	if nginxCfg.Keepalive != "" {
+		keepalive = nginxCfg.Keepalive
+	}
+
+	minions := mergeableIngs.Minions
+	for _, minion := range minions {
+		pems, jwtKeyFileName := cnf.updateSecrets(&minion)
+		nginxCfg := cnf.generateNginxCfg(&minion, pems, jwtKeyFileName)
```

Comment:
I suggest to replace all minions annotations with the annotations from the master here.

Reply:
Done, but we will eventually need to change this logic.